lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -432,6 +432,10 @@ def confirm_context_exists
     end
 
     def confirm_cluster_reachable
+      if server_version < Gem::Version.new(MIN_KUBE_VERSION)
```

Comment:
You'll want to do this _after_ the rest of the code in this method; the rest of the code confirms that the server actually exists, so we should do that before trying to ask it what its version is.